src/components/SearchResults/SearchFilterList/SearchFilter.js

```diff
@@ -26,54 +26,77 @@ const ToggleButton = styled(sharedButton)`
 export const SearchFilter = ({ filterGroup, filterOption, filterLabel }) => {
   const { viewport } = useResponsive()
   const { isFilterChecked, toggleFilter } = useSearchManager()
+  const filterList = Object.entries(filterGroup).sort((a, b) => b[1] - a[1])
+  const filterListCount = filterList.length
   const maxCount = 5
-  const filterList = Object.entries(filterGroup)
-  const filterLength = filterList.length
-  const [userInput, setUserInput] = useState('')
+  const isMoreThanMaxCount = filterListCount > maxCount
+  const [filteredResults, setFilteredResults] = useState(
+    filterList.slice(0, maxCount)
+  )
   const [open, setOpen] = useState(false)
-  const formattedFilterList = filterList
-    .filter((option) =>
-      formatString(option[0]).toLowerCase().startsWith(userInput.toLowerCase())
+  const [userInput, setUserInput] = useState('')
+
+  const handleToggleFilterList = (val) => {
+    setUserInput(val)
+    setFilteredResults(() =>
+      // eslint-disable-next-line no-nested-ternary
+      val.trim() !== ''
+        ? filterList.filter((option) =>
+            (filterOption === 'platform'
+              ? formatPlatformName(cache.platforms[option[0]]) || option[0]
+              : formatString(option[0])
+            )
+              .toLowerCase()
+              .includes(val.toLowerCase())
+          )
+        : open
+        ? filterList
+        : filterList.slice(0, maxCount)
```

Comment:
This is a bit much. Can we break this up to determine what to set instead of having all of the logic in this function?